cpp/ql/lib/semmle/code/cpp/ir/implementation/raw/internal/reachability/ReachableBlock.qll

```diff
@@ -10,6 +10,65 @@ predicate isInfeasibleInstructionSuccessor(Instruction instr, EdgeKind kind) {
   or
   instr.getSuccessor(kind) instanceof UnreachedInstruction and
   kind instanceof GotoEdge
+  or
+  isCallToNonReturningFunction(instr)
+}
+
+/**
+ * Holds if all calls to `f` never return (e.g. they call `exit` or loop forever)
+ */
+private predicate isNonReturningFunction(IRFunction f) {
```

Comment:
I see that Wireshark has a 7% performance regression with this change. From previous experiments I know that Wireshark has quite a few very very large `IRFunction`s generated for initialization of global variables and static locals. Such initializers probably never unconditionally call a non-returning function so it may performance-wise be a win to just always assume that if `f.getFunction() instanceof Declaration` then `f` always returns.

Reply:
Actually this was due to a CP in `isInfeasibleInstructionSuccessor` as `EdgeKind` was entirely unrestricted. `EdgeKind` isn't that big as it is the number of distant integers used in `case` (+ O(1)) so the CP went unnoticed on most dbs.